### ISSUE TITLE
error TS2339: Property 'migrateSchema' does not exist on type

### ISSUE THREAD

**Issue by sadranyi** [strongloop/loopback-next#5608]
## Steps to reproduce
After going through the getting started steps at the link below
[Getting Started](https://loopback.io/getting-started.html)

run `npm start` 
## Current Behavior
returns `src/migrate.ts:9:13 - error TS2339: Property 'migrateSchema' does not exist on type 'XXXXApplication'.`

## Expected Behavior

Starter project with a default **ping** route on localhost port 3000 `http://127.0.0.1:3000/ping`

## Additional information
win32 x64 14.3.0

## Related Issues

_See [Reporting Issues](http://loopback.io/doc/en/contrib/Reporting-issues.html) for more tips on writing good issues_

**Comment by dhmlau** [comment]
@sadranyi, were you trying to run `npm run migrate` as well? By default, `npm start` won't call the migrate script. 
Anyway, please remember to build before running the migrate script, i.e.
```
$ npm run build
$ npm run migrate
```

Hope it helps.

**Comment by sadranyi** [comment]
worked like magic. should we always have to do this each time we create a new app?

**Comment by dhmlau** [comment]
The `npm run migrate` does not transpile the code, so everytime you make a change, you'd need to run `npm run build` first. 

I've created https://github.com/strongloop/loopback-next/issues/4189 to propose to include running the build.

**Comment by sadranyi** [comment]
but seem the build script is run as part of `npm start`

**Comment by dhmlau** [comment]
> It looks like the build script is run as part of npm start

Yes.

If you agree, I'll close this issue. Please feel free to continue the discussion of whether to include the build script as part of the migrate script. :). Thanks!

**Comment by sadranyi** [comment]
yes please close it. thanks

**Comment by rohit-ambre** [comment]
Hi,
I had a same issue while running application like explained in getting started.
I also ran `npm run build` it still throws the same error as 
`src/migrate.ts:9:13 - error TS2339: Property 'migrateSchema' does not exist on type 'Loopback4TestApplication'.`
I also tried running `npm run migrate` and gives following error
`Migrating schemas (alter existing schema)
Cannot migrate database schema TypeError: app.migrateSchema is not a function`

I'm using Windows 10, node 12.16.1

**Comment by andersonaskm** [comment]
Hello, I´m having the same problem. Looking inside the generated source code, it shows the following lines

**const app = new LoopbackTesteApplication();
  await app.boot();
  await app.migrateSchema({existingSchema}); // my vscode shows an error that it cannot find this method**

And looking inside the generated "LoopbackTesteApplication", there is no migrateSchema method there. Above is the generated code.

**import {BootMixin} from '@loopback/boot';
import {ApplicationConfig} from '@loopback/core';
import {
  RestExplorerBindings,
  RestExplorerComponent,
} from '@loopback/rest-explorer';
import {RestApplication} from '@loopback/rest';
import path from 'path';
import {MySequence} from './sequence';

export {ApplicationConfig};

export class LoopbackTesteApplication extends BootMixin(RestApplication) {
  constructor(options: ApplicationConfig = {}) {
    super(options);

    // Set up the custom sequence
    this.sequence(MySequence);

    // Set up default home page
    this.static('/', path.join(__dirname, '../public'));

    // Customize @loopback/rest-explorer configuration here
    this.configure(RestExplorerBindings.COMPONENT).to({
      path: '/explorer',
    });
    this.component(RestExplorerComponent);

    this.projectRoot = __dirname;
    // Customize @loopback/boot Booter Conventions here
    this.bootOptions = {
      controllers: {
        // Customize ControllerBooter Conventions here
        dirs: ['controllers'],
        extensions: ['.controller.js'],
        nested: true,
      },
    };
  }
}**

**Comment by achrinza** [comment]
RepositoryMixin is needed as part of RestApplication:


https://loopback.io/doc/en/lb4/apidocs.repository.repositorymixindoc.html

**Comment by etvoorde** [comment]
The getting started page lists the following settings:

[?] Select project build settings: Enable tslint, Enable prettier, Enable mocha, Enable loopbackBuild

I believe it should include "repositories" as well.

**Comment by achrinza** [comment]
> The getting started page lists the following settings:
> 
> [?] Select project build settings: Enable tslint, Enable prettier, Enable mocha, Enable loopbackBuild
> 
> I believe it should include "repositories" as well.

Seems like we may not be looking at the same thing.

Here's what I see in the [Getting Started](https://loopback.io/doc/en/lb4/Getting-started.html#create-a-new-project) docs:

![image](https://user-images.githubusercontent.com/25147899/85144217-73b32d80-b27d-11ea-945e-0566dfd4188c.png)

The `repositories` feature is enabled.

---

The latest CLI version also has all the features enabled by default:

![image](https://user-images.githubusercontent.com/25147899/85144641-081d9000-b27e-11ea-98a3-013c2993a4ee.png)

**Comment by andersonaskm** [comment]
Hi, I´ve just selected them all and the problem remains the same (several problems by the way)


![image](https://user-images.githubusercontent.com/8411131/85144770-01612f80-b222-11ea-8958-6dba47536c1c.png)

![image](https://user-images.githubusercontent.com/8411131/85144817-18078680-b222-11ea-9e92-4920494d3217.png)

**Comment by etvoorde** [comment]
I was looking at the following page: https://loopback.io/getting-started.html

**Comment by achrinza** [comment]
> I was looking at the following page: https://loopback.io/getting-started.html

@etvoorde Thanks for raising the issue, I've opened https://github.com/strongloop/loopback.io/pull/997 to update the page.

---

@andersonaskm I've replied in #5772. Let's continue the conversation there.

**Comment by lucasbagatini** [comment]
I had the same problem using yarn. After installing everything again but using npm it worked.

**Comment by StanisalvPopadynets** [comment]
I had the same problem. Tried everything once again but this time I left all features for the project selected by default and it worked on the first try, server is running.

**Comment by NorthDecoder** [comment]
If you are here reading this closed issue, like I am, you are having the same problems noted in several comments above.  My difficulty came from reading the scaffolding instructions exactly, _"Select features to enable in the project. ( Press , 'space' to select, 'a' to toggle all, 'i' to invert selection )"_.  I assumed that meant I needed to enable the features.
![01-lb4-scaffold-form-default](https://user-images.githubusercontent.com/9598008/96386870-0c39bb80-1153-11eb-9732-cc1819a675f2.PNG)

----

Therefore I dutifully arrowed to each feature and pressed the space bar to enable the feature.
![02-lb4-scaffold-form-toggle](https://user-images.githubusercontent.com/9598008/96387072-68510f80-1154-11eb-839a-56bda62604f0.PNG)

----
Noting  @StanisalvPopadynets comment above, I carefully looked at my selections ( after wiping out old install, then reinstalling ) and I see there is a green rectangle and a capital 'C' indicating status.  Because my interface does not look like the installation instructions in the docs I had to guess the status and guessed wrong.

In my case the 'green rectangle' means enabled and I suppose the 'C' means cleared.

----

> My conclusion is that __all features are enabled by default__ and there is nothing to do but continue by pressing 'enter'.

----

It might make sense for the wording to be different, for example:
__FROM__: "Select features to enable in the project. "
__TO__: "Select features to __disable__ in the project."
but it is kind of splitting hairs and maybe confusing either way.

HTH
